android/src/main/java/io/radar/react/RNRadarReceiver.java

```diff
@@ -18,36 +19,35 @@
 import io.radar.sdk.RadarReceiver;
 import io.radar.sdk.model.RadarEvent;
 import io.radar.sdk.model.RadarUser;
-import java.util.concurrent.atomic.AtomicInteger;
 
+/**
+ * Implementation of {@link RadarReceiver} that relays events received on Android to React Native
+ * code
+ */
 public class RNRadarReceiver extends RadarReceiver {
 
-    private ReactNativeHost reactNativeHost;
-    private PendingResult result;
-    private AtomicInteger pendingCount = new AtomicInteger(0);
     private static final String TAG = "RNRadarReceiver";
 
     private void invokeSendEvent(ReactContext reactContext, String eventName, Object data) {
-        reactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit(eventName, data);
+        reactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                .emit(eventName, data);
     }
 
-    private void sendEvent(final String eventName, final Object data) {
+    private void sendEvent(ReactNativeHost reactNativeHost,
+                           final String eventName,
+                           final Object data) {
         final ReactInstanceManager reactInstanceManager = reactNativeHost.getReactInstanceManager();
         ReactContext reactContext = reactInstanceManager.getCurrentReactContext();
         if (reactContext == null) {
-            if (result == null) {
-                result = goAsync();
-            }
-            pendingCount.incrementAndGet();
-            reactInstanceManager.addReactInstanceEventListener(new ReactInstanceManager.ReactInstanceEventListener() {
+            reactInstanceManager.addReactInstanceEventListener(
+                    new ReactInstanceManager.ReactInstanceEventListener() {
+
```

Comment:
Are there any potential gotchas from not using a `PendingResult` anymore?

Was reading these:
https://developer.android.com/reference/android/content/BroadcastReceiver.PendingResult

@brettguenther or @nickpatrick may know more of the back story here
https://stackoverflow.com/questions/22741202/how-to-use-goasync-for-broadcastreceiver

Reply:
Here's the original PR and Github issue where the `PendingResult` was added:

https://github.com/radarlabs/react-native-radar/pull/36
https://github.com/radarlabs/react-native-radar/issues/35

Reply:
`goAsync()` is a method on a BroadcastReceiver to ensure that long operations are not killed by the OS. Since this class no longer extends BroadcastReceiver, there's no PendingIntent to use.

Reply:
Hmm, yeah good point.  Do you think this release will cause a regression for https://github.com/radarlabs/react-native-radar/issues/35, or does the fact that the new code path happens not a `BroadcastReciever` make it so this won't happen?

Reply:
From the stack trace on [the issue](https://github.com/radarlabs/react-native-radar/issues/35), and with the understanding that the broadcast receiver is killed by Android OS if its operation takes too long, it is clear to me that the issue was caused when react had not yet been fully initialized, and the listener was added within the receiver such that its `onReactContextInitialized` callback would be notified when react was ready. This would trigger an event and cause the NPE because the BroadcastReceiver had been garbage collected. I am fairly confident that this is no longer a concern, because the receiver will not be garbage collected by the OS anymore.